sopel/cli/run.py

```diff
@@ -0,0 +1,658 @@
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel stop` instead)"))
+    parser.add_argument("-k", '--kill', action="store_true", dest="kill",
+                        help=(
+                            "Kill Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel stop --kill` instead)"))
+    parser.add_argument("-r", '--restart', action="store_true", dest="restart",
+                        help=(
+                            "Restart Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel restart` instead)"))
+    parser.add_argument("-l", '--list', action="store_true",
+                        dest="list_configs",
+                        help="List all config files found")
+    parser.add_argument("-m", '--migrate', action="store_true",
+                        dest="migrate_configs",
+                        help="Migrate config files to the new format")
+    parser.add_argument('--quiet', action="store_true", dest="quiet",
+                        help="Suppress all output")
+    parser.add_argument('-w', '--configure-all', action='store_true',
+                        dest='wizard',
+                        help=(
+                            "Run the configuration wizard "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel configure` instead)"))
+    parser.add_argument('--configure-modules', action='store_true',
+                        dest='mod_wizard',
+                        help=(
+                            "Run the configuration wizard, but only for the "
+                            "module configuration options "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel configure --modules` instead)"))
+    parser.add_argument('-v', action="store_true",
+                        dest='version_legacy',
+                        help=(
+                            "Show version number and exit "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use -V/--version instead)"))
+    parser.add_argument('-V', '--version', action='store_true',
+                        dest='version',
+                        help='Show version number and exit')
+
+
+def build_parser():
+    """Build an ``argparse.ArgumentParser`` for the bot"""
+    parser = argparse.ArgumentParser(description='Sopel IRC Bot',
+                                     usage='%(prog)s [options]')
+    subparsers = parser.add_subparsers(
+        title='sub-commands',
+        description='List of Sopel\'s sub-commands',
+        dest='action')
+
+    # manage `legacy` sub-command
+    parser_legacy = subparsers.add_parser(
+        'legacy', help='Launch Sopel\'s legacy command line')
```

Comment:
I had a thought back when I had a chance to play with this a bit in an actual terminal:

Can the `legacy` sub-command be hidden from the list of choices given by `--help`? I wasn't confused by its presence because I'd read some of the PR already, but to a user it makes no sense to have this as a choice. It's an internal implementation detail, really.

Reply:
Fixed, nice find.